tests/modules/users/repository/user.repository.spec.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  createUser,
+  fetchUser,
+  getExistingUser
+} from '@modules/users/repository/user.repository';
+import { User } from '@prisma/client';
+import { prismaMock } from '@tests/prismaTestSetup';
+
+const mockDataStore = [
+  {
+    id: '1',
+    firstName: 'John',
+    lastName: 'Doe',
+    email: 'john.doe@example.com',
+    phoneNumber: '123-456-7890',
+    address: '123 Main St, Anytown USA',
+    isActive: true,
+    isVerified: true,
+    createdAt: new Date(),
+    updatedAt: new Date()
+  },
+  {
+    id: '2',
+    firstName: 'Jane',
+    lastName: 'Doe',
+    email: 'jane.doe@example.com',
+    phoneNumber: '987-654-3210',
+    address: '456 Oak St, Anytown USA',
+    isActive: false,
+    isVerified: true,
+    createdAt: new Date(),
+    updatedAt: new Date()
+  }
+] as User[];
+
+describe('createUser function', () => {
+  const mockUserData = {
+    firstName: 'John',
+    lastName: 'Doe',
+    email: 'john.doe@example.com',
+    phoneNumber: '123-456-7890',
+    address: '123 Main St, Anytown USA',
+    isActive: true,
+    isVerified: true,
+    password: 'password123'
+  };
+
+  //@ts-ignore
+  const mockCreate = prismaMock.user.create;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should call prisma.user.create with the correct arguments', async () => {
+    //Arrange
+    const expectedUserData = mockDataStore[0];
+    mockCreate.mockResolvedValueOnce(expectedUserData);
+    //Act
+    const result = await createUser(mockUserData);
+    //Assert
+    expect(mockCreate).toHaveBeenCalledWith({ data: mockUserData });
+    expect(result).toEqual(expectedUserData);
+  });
+
+  test('should throw an error if prisma.user.create throws an error', async () => {
+    //Arrange
+    const errorMessage = 'Failed to create user';
+    mockCreate.mockRejectedValueOnce(new Error(errorMessage));
+    //Act
+    const result = createUser(mockUserData);
+    //Assert
+    await expect(result).rejects.toThrow(errorMessage);
+  });
+});
+
+describe('fetchUser function', () => {
+  const mockFindMany = prismaMock.user.findMany;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should call prisma.user.findMany and return the user data', async () => {
```

Comment:
Each test block should be independent of other test block. Since no data is saved in the DB during this test, so this test should return empty array instead of the data.

Reply:
We have mocked the return data for this already, returning mocked empty array or the mocked user array shouldn't make any difference, after all its mocked data to be tested. :thinking: